README.md

```diff
@@ -121,15 +121,75 @@ but it doesn't traverse the object recursively converting nested items:
 ```
 
 
-## Public API
+## API
+
+### Extending `itemadapter`
+
+This package allows to handle arbitrary item classes, by implementing an adapter interface:
+
+_class `itemadapter.adapter.AdapterInterface(item: Any)`_
+
+Abstract Base Class for adapters. An adapter that handles a specific type of item should
+inherit from this class and implement the abstract methods defined on it. `AdapterInterface`
+inherits from [`collections.abc.MutableMapping`](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping),
+so all methods from the `MutableMapping` class should be implemented as well.
+
+* _class method `is_item(item: Any) -> bool`_
+
+    Return `True` if the adapter can handle the given item, `False` otherwise
+
+* _method `get_field_meta(self, field_name: str) -> MappingProxyType`_
+
+    Return metadata for the given field name, if available
+
+* _method `field_names(self) -> KeysView`_:
+
+    Return a dynamic view of the item's field names
+
+* _method `asdict(self) -> dict`_:
+
+    Return a dictionary containing the contents of the adapted item, converting nested structures as well
```

Comment:
Here an example of using the itemadapter API to achieve this could be really useful. In fact, I wonder if we can make this trivial to implement with a helper function.

Maybe instead of a method that converts nested structures as well, adapter interfaces could simply implement top-level conversion to dict, and let item adapter handle recursive conversion. Or offer both options, let users implement either `asdict` or something for top-level conversion, and let the default `asdict` use the custom top-level implementation when available.

Reply:
Right, I'm checking this and it seems like it's not necessary to require the manual implementation of `asdict`. The `MutableMapping` interface already provides a way to do a top-level conversion, and then the `ItemAdapter` class itself can inspect the object recursively to keep converting.
Thanks for the suggestion, it does make the process simpler.